internal/pkg/service/buffer/api/service/service.go

```diff
@@ -40,6 +44,9 @@ func (*service) HealthCheck(dependencies.ForPublicRequest) (res string, err erro
 	return "OK", nil
 }
 
+// nolint:godox
+// TODO: collect errors instead of bailing on the first one
+
 func (*service) CreateReceiver(d dependencies.ForProjectRequest, payload *buffer.CreateReceiverPayload) (res *buffer.Receiver, err error) {
```

Comment:
Tahle metoda je tady dost obrovská. Stejně tak i `GetReceiver`. Mohli byste poradit, jak to nejlíp rozdělit? Něco bych vymyslel, ale zajímá mě, jak byste to udělali vy.

Reply:
- spravil by som si nejake dalsie subory, napr. `map_receiver.go`
- tam by som cisto mapoval `payload -> model` a `model -> payload`
- vzhladom na transakcie, budes potom na tomto miesto volat iba jednu metodu, `configstore.CreateReceiver`.
- mozno budes potrebovat doplnit dalsie `types`, napr. `ReceiverWithExports` (alebo `ReceiverDetail`), podobne `ExportWithMapping`/`ExportDetail`.
- zide sa ti to aj pri generovani `list/get` odpovedi, kde to potrebujes mat spojene

- tj. podel si to na vrstvy , dataflow: `payload -> mapper -> configstore -> etcd`
- a opacne `etcd -> configstore -> mapper -> response`.

Reply:
- ten `mapper` moze byt aj novy balicek
- `service` a `configstore` si mozes rozdelit na viacero suborov, `receiver.go`, `export.go`, ...